### PR TITLE
Libtk205 build

### PR DESCRIPTION
**Pull request by tanaya-bigladder**

### CODE REVIEW

**Review by chipbarnaby — APPROVED** [review]

File: libtk205/include/libtk205.h [hunk 1]
```diff
@@ -0,0 +1,38 @@
...
+#include "RS0003.h"
+#include "RS0004.h"
+#include "RS0005.h"
+#include "RS0006.h"
+#include "RS0007.h"
+#include "nlohmann/json.hpp"
+
+namespace libtk205_NS {
+
+    class A205_SDK
```
Comment (nealkruis) [inline libtk205/include/libtk205.h:18]: Do we need this class? I wonder if these can all be part of a single data file base class that all the "root" classes inherit from.

File: libtk205/include/libtk205.h [hunk 2]
```diff
@@ -0,0 +1,38 @@
...
+#include "RS0001.h"
+#include "RS0002.h"
+#include "RS0003.h"
+#include "RS0004.h"
+#include "RS0005.h"
+#include "RS0006.h"
+#include "RS0007.h"
+#include "nlohmann/json.hpp"
+
+namespace libtk205_NS {
```
Comment (nealkruis) [inline libtk205/include/libtk205.h:16]: Can we eliminate this namespace level in favor of `ASHRAE205_NS`?

File: libtk205/src/libtk205.cpp [hunk 3]
```diff
@@ -0,0 +1,134 @@
...
+        Validator validator;
+        NlohmannJsonAdapter input_adapter(input_doc);
+        if (!validator.validate(schema, input_adapter, NULL))
+        {
+            throw std::runtime_error("Validation failed.");
+        }
+        return true;
+    }
+
+    RS0001_NS::RS0001 A205_SDK::Load_RS0001(const char* input_file)
```
Comment (nealkruis) [inline libtk205/src/libtk205.cpp:49]: These functions should all be generated by a template for each "root" data group.

File: libtk205/include/libtk205.h [hunk 4]
```diff
@@ -0,0 +1,38 @@
+#ifndef LIBTK205_H_
+#define LIBTK205_H_
+
+#include <vector>
+#include <memory>
+#include "ASHRAE205.h"
```
Comment (nealkruis) [inline libtk205/include/libtk205.h:6]: This file should be generated from a template to automatically include the header file corresponding to each schema file.

File: libtk205/include/libtk205.h (hunk 2 above)
Comment (tanaya-bigladder) [reply libtk205/include/libtk205.h:16]: Makes sense. Will do.

File: libtk205/include/libtk205.h (hunk 1 above)
Comment (nealkruis) [reply libtk205/include/libtk205.h:18]: It's not necessarily related. I was thinking that some of this functionality could be moved to the base class (e.g., read_binary_file).

**Comment by tanaya-bigladder** [comment]
Superseded by [44](https://github.com/open205/toolkit-205/pull/44/files).

**Pull request closed without merging**